lib/good_job/metrics.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+module GoodJob # :nodoc:
+  # Metrics for the scheduler.
+  class Metrics
+    def initialize
```

Comment:
@bensheldon What do you think about making metrics more generic? something like
```suggestion
    def initialize(*keys)
      @counters = keys.map { |key| [key, Concurrent::AtomicFixnum.new] }
```
then increment could be something like
```
def increment(key)
@keys.fetch(key).increment
end
```

Reply:
My preference (at the moment) is to keep them hardcoded. They are still fairly discrete and I also worry about fat-fingering them when typing (having explicit methods will at least raise an exception)